### PR TITLE
DEPR: Enforce deprecation of dropping columns when numeric_only=False in groupby / resample

### PR DESCRIPTION
**Pull request by rhshadrach**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

Prior to calling the cython code, cummin/cummax/cumsum/cumprod raise NotImplementedError on object dtypes. The Series path intercepts this and raises a TypeError instead. I've implemented the same here ~except for DataFrame, we can't readily tell which dtype(s) would fail, so it raises a somewhat vague "at least one dtype" message~.

### CODE REVIEW

File: pandas/core/groupby/generic.py [hunk 1]
```diff
@@ -1357,7 +1357,13 @@ def arr_func(bvalues: ArrayLike) -> ArrayLike:
 
         # We could use `mgr.apply` here and not have to set_axis, but
         #  we would have to do shape gymnastics for ArrayManager compat
-        res_mgr = mgr.grouped_reduce(arr_func, ignore_failures=True)
+        try:
+            res_mgr = mgr.grouped_reduce(
+                arr_func, ignore_failures=numeric_only is lib.no_default
+            )
+        except NotImplementedError as err:
+            msg = f"{how} is not supported for at least one provided dtype"
```
Comment (jbrockmendel) [inline pandas/core/groupby/generic.py:1365]: why not just let the original exception surface?  wont that include dtype information?

Comment (rhshadrach) [reply pandas/core/groupby/generic.py]: Thanks! That's much better.

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @rhshadrach

File: pandas/core/groupby/generic.py [hunk 2]
```diff
@@ -1357,7 +1357,13 @@ def arr_func(bvalues: ArrayLike) -> ArrayLike:
 
         # We could use `mgr.apply` here and not have to set_axis, but
         #  we would have to do shape gymnastics for ArrayManager compat
-        res_mgr = mgr.grouped_reduce(arr_func, ignore_failures=True)
+        try:
+            res_mgr = mgr.grouped_reduce(
+                arr_func, ignore_failures=numeric_only is lib.no_default
+            )
+        except NotImplementedError as err:
+            # For NotImplementedError, args[0] is the error message
```
Comment (jbrockmendel) [inline pandas/core/groupby/generic.py:1365]: is this still the case if it was raised with just `raise NotImplementedError`?

File: doc/source/whatsnew/v2.0.0.rst [hunk 3]
```diff
@@ -557,6 +557,7 @@ Removal of prior version deprecations/changes
 - Enforced deprecation ``numeric_only=None`` (the default) in DataFrame reductions that would silently drop columns that raised; ``numeric_only`` now defaults to ``False`` (:issue:`41480`)
 - Changed default of ``numeric_only`` to ``False`` in all DataFrame methods with that argument (:issue:`46096`, :issue:`46906`)
 - Changed default of ``numeric_only`` to ``False`` in :meth:`Series.rank` (:issue:`47561`)
+- Enforced deprecation of silently dropping nuisance columns in groupby and resample operations when ``numeric_only=False`` (:issue:`41475`)
```
Comment (jbrockmendel) [inline doc/source/whatsnew/v2.0.0.rst:560]: maybe this is different from the issue i recall, but i thought the silent-dropping was with numeric_only=None, and we're just getting rid of that altogether?

File: pandas/core/groupby/generic.py (hunk 2 above)
Comment (rhshadrach) [reply pandas/core/groupby/generic.py:1365]: Ah - no, it's not. I'll fix in a followup. Thanks.

File: doc/source/whatsnew/v2.0.0.rst (hunk 3 above)
Comment (rhshadrach) [reply doc/source/whatsnew/v2.0.0.rst:560]: Right - I've been trying to figure out how to split this up into manageable pieces. Certain ops in 1.x would drop nuisance columns even when numeric_only=False, and others would only do it when numeric_only=None. It seemed easiest to handle the False case first, then get rid of the None.

Comment (jbrockmendel) [reply doc/source/whatsnew/v2.0.0.rst:560]: cool, thanks

**Pull request merged**